src/any_llm/utils/messages_compat.py

```
@@ -165,13 +217,83 @@ def _convert_assistant_blocks_to_openai(blocks: list[dict[str, Any]]) -> list[di
+    """
+    source = block.get("source", {})
+    source_type = source.get("type")
+    if source_type == "text":
+        return {"type": "text", "text": source.get("data", "")}
+    if source_type == "base64":
+        media_type = source.get("media_type", "application/pdf")
+        return {"type": "file", "file": {"file_data": f"data:{media_type};base64,{source.get('data', '')}"}}
+    return {"type": "file", "file": {"file_data": source.get("url", "")}}
+
+
+def _convert_tool_result_content(tool_content: Any) -> tuple[str, list[dict[str, Any]]]:
```

Ruff (0.16.1)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `tool_content`

(ANN401)